lib/composer/package/loader/project_attributes_loader.rb

```diff
@@ -0,0 +1,56 @@
+module Composer
+  module Package
+    module Loader
+      # Loads a package from the project attributes
+      class ProjectAttributesLoader < Composer::Package::Loader::ProjectLoader
+
+        def initialize(loader = nil)
+          if loader.nil?
+            loader = Composer::Package::Loader::HashLoader.new
+          end
+          @loader = loader
+        end
+
+        # Load a json string or file
+        # Param:  Project project The target gitlab project.
+        # Param:  Branch|Tag ref The target gitlab project branch/tag.
+        # Param:  Hash config A hash containing addional package configuration.
+        # Returns: Composer::Package::Package
+        def load(project, ref, type = 'library')
+          config = {
+            'name'                => project.path_with_namespace.gsub(/\s/, '').downcase,

```

Comment:
@DouweM Since the names are guaranteed to be space-less then I will remove the unnecessary gsub and keep the downcase. 

Also let me point out that the package name doesn't need to match the path of the Gitlab repo and can be arbitrarily set using the custom json option to any namespace/package_name.

The "source" and "dist" keys are used to fetch the package from the correct project url.